src/api/app/models/user.rb

```diff
@@ -218,23 +218,28 @@ def self.find_with_credentials_via_ldap(login, password)
 
   # The currently logged in user (might be nil). It's reset after
   # every request and normally set during authentification
+  # DEPRECATED! Prefer possibly_nobody or required
   def self.current
     Thread.current[:user]
   end
 
-  def self.nobody
-    Thread.current[:nobody] ||= find_nobody!
-  end
-
   # Currently logged in user or nobody user if there is no user logged in.
   # Use this to check permissions, but don't treat it as logged in user. Check
   # is_nobody? on the returned object
-  def self.current_or_nobody
+  def self.possibly_nobody
     current || nobody
   end
 
-  def self.session=(user)
-    Thread.current[:user] = user
+  # Currently logged in user. Will thrown an exception if no user is logged in.
+  # So the controller needs to require login if using this (or models using it)
+  def self.session!
+    raise ArgumentError, 'Requiring user, but found nobody' unless session
+    current
+  end
```

Comment:
Could we move this to to the pundit policies right away? The method doesn't seem to be used right now anyway. So this wouldn't require to much effort, right?

And on second thought... did this slip in? Given that it's unused...

Reply:
Guys, you're killing me. When I sent a PR with the new API and 220 files using it, it was too big for you. Now I split the new API so you can review it in a special PR and now you're complaining the new API is unused?

I quote from a comment of 9 days ago: 
> On how to use these functions, look at #7219

That comment is slightly outdated now as I renamed User.required to User.session!, but it's still the same. Moving this to pundit policies is a per usage effort

As a matter of fact, many controllers already use `before_login :require_login`, so the User.session! usage there is pointless, but can be useful for the models used.

Reply:
pundit? you have to explain that to me please

Reply:
> Guys, you're killing me. 

I understand your frustration. Please keep in mind that this is a big change, it's not so easy to follow.

Reply:
> When I sent a PR with the new API and 220 files using it, it was too big for you

I reviewed that one as well :wink:

> and now you're complaining the new API is unused?

My ~~complain~~ question was if we could already move such things into pundit.
And given that this, and the other PR, are just interim changes it's natural that such questions come up:-)